universal-application-tool-0.0.1/app/repository/QuestionRepository.java

```diff
@@ -98,16 +102,24 @@ public Question updateOrCreateDraft(QuestionDefinition definition) {
     }
   }
 
+  /**
+   * Update DRAFT and ACTIVE questions that reference {@code oldEnumeratorId} to reference {@code
+   * newEnumeratorId}.
+   */
   private void updateAllRepeatedQuestions(long newEnumeratorId, long oldEnumeratorId) {
+    // TODO: This seems error prone as a question could be present as a DRAFT and ACTIVE.
+    // Investigate further.
```

Comment:
yeah this is a bit suspicious to me since an invariant is supposed to be that only draft questions and programs are mutable

Reply:
When processing the ACTIVE ones it'll make new DRAFTS if necessary, the problem is that it might process an EXISTING draft revision, THEN the ACTIVE revision which, which might blow away the draft?  unsure, but doesn't seem right.

There's a similarly structured method that excludes the drafts when processing the actives, so I think that's an oversight here, I'll resolve subsequently.